src/QuickMarcEditor/QuickMarcEditorRows/ControlNumberField/ControlNumberField.js

```diff
@@ -39,45 +37,59 @@ const ControlNumberField = ({
   marcType,
   action,
   recordRows,
+  onChangeControlNumberRecord,
 }) => {
   const intl = useIntl();
-  const { input } = useField(name);
   const { getAuthorityFileNextHrid, isLoading: isLoadingHrid } = useAuthorityFileNextHrid();
 
-  const handleChangeContent = input.onChange;
+  const [selectedSource, setSelectedSource] = useState('');
+
   const contentOf010Row = recordRows.find(row => row.tag === '010')?.content;
   const valueOf010$a = getContentSubfieldValue(contentOf010Row).$a?.[0];
 
-  const [selectedSource, setSelectedSource] = useState('');
+  const controlNumberIndex = recordRows.findIndex(row => row.tag === '001');
+
+  const handleChangeControlNumberRecord = useCallback((content, sourceFile) => {
+    const controlNumberRow = recordRows[controlNumberIndex];
+
+    onChangeControlNumberRecord({
+      index: controlNumberIndex,
+      field: {
+        ...controlNumberRow,
+        content,
+      },
+      sourceFile: sourceFile || controlNumberRow._sourceFile,

```

Comment:
I suggest another approach to store selected source file - we can leverage Context for this task.
There can be only one 001 field, so there can be one selected source file that defines the MARC record's HRID
If we use context, we can change its value directly from `<ControlNumberField>` without having to store it in form state, or pass it up the component tree, and we can get the conext value in `useValidation` to pass it to validators.

I'm thinking we can make it generic like `QuickMarcConext`, so we can store other global values, like `marcType`, `action`, `instance`, maybe `initialValues`